common/type_system/TypeSpec.h

```diff
@@ -149,4 +150,9 @@ class TypeSpec {
   // arguments (most of the time) and makes the type analysis pass in the decompiler 2x faster.
   std::vector<TypeSpec>* m_arguments = nullptr;
   cu::SmallVector<TypeTag, 1> m_tags;
+
+  // definition information
+  std::string m_defining_file;
+  int m_line_number;
+  int m_line_offset;
```

Comment:
Ah, I agree -- but this looks like a left-over from when i first started the work.  Since im only doing decompiler use-cases for now, this isn't needed for now (the method that sets these vars isn't even called).

Putting it on the `Type` makes more sense, Somewhat related -- I'd like to see more fields as well, mainly the type/functions docstring.  If this is set in `all-types.gc` then we can carry it forward to the output of the decompiler and the LSP can show the docstring on hover information.